auth-service/src/test/kotlin/uk/gov/communities/delta/dbintegration/PasswordTokenServiceTest.kt

```diff
@@ -0,0 +1,213 @@
+        result = registrationSetPasswordTokenService.validateToken(token, userCN)
+        assertTrue(result is PasswordTokenService.NoSuchToken)
+    }
+
+    @Test
+    fun testLookupForResetWithDelete() = testSuspend {
+        val token = resetPasswordTokenService.createToken(userCN)
+        var result = resetPasswordTokenService.consumeToken(token, userCN)
+        assertTrue(result is PasswordTokenService.ValidToken)
+        result = resetPasswordTokenService.validateToken(token, userCN)
+        assertTrue(result is PasswordTokenService.NoSuchToken)
+    }
+
+    @Test
+    fun testLookupForSetWithoutDelete() = testSuspend {
+        val token = registrationSetPasswordTokenService.createToken(userCN)
+        var result = registrationSetPasswordTokenService.validateToken(token, userCN)
+        assertTrue(result is PasswordTokenService.ValidToken)
+        result = registrationSetPasswordTokenService.validateToken(token, userCN)
+        assertTrue(result is PasswordTokenService.ValidToken)
+    }
+
+    @Test
+    fun testLookupForResetWithoutDelete() = testSuspend {
+        val token = resetPasswordTokenService.createToken(userCN)
+        var result = resetPasswordTokenService.validateToken(token, userCN)
+        assertTrue(result is PasswordTokenService.ValidToken)
+        result = resetPasswordTokenService.validateToken(token, userCN)
+        assertTrue(result is PasswordTokenService.ValidToken)
+    }
+
+    @Test
+    fun testOnlyOneTokenPerUserForSet() = testSuspend {
+        val originalToken = registrationSetPasswordTokenService.createToken(userCN)
+        val replacementToken = registrationSetPasswordTokenService.createToken(userCN)
+        var result = registrationSetPasswordTokenService.validateToken(originalToken, userCN)
+        assertTrue(result is PasswordTokenService.NoSuchToken)
+        result = registrationSetPasswordTokenService.consumeToken(replacementToken, userCN)
+        assertTrue(result is PasswordTokenService.ValidToken)
+    }
+
+    @Test
+    fun testOnlyOneTokenPerUserForReset() = testSuspend {
+        val originalToken = resetPasswordTokenService.createToken(userCN)
+        val replacementToken = resetPasswordTokenService.createToken(userCN)
+        var result = resetPasswordTokenService.validateToken(originalToken, userCN)
+        assertTrue(result is PasswordTokenService.NoSuchToken)
+        result = resetPasswordTokenService.consumeToken(replacementToken, userCN)
+        assertTrue(result is PasswordTokenService.ValidToken)
+    }
+
+    @Test
+    fun testLookupExpiredTokenForSet() = testSuspend {
+        val token = registrationSetPasswordTokenService.createToken(userCN)
+        withContext(Dispatchers.IO) {
+            testDbPool.useConnectionBlocking("Test make set password token expire") {
+                val stmt = it.prepareStatement(
+                    "UPDATE set_password_tokens SET created_at = ? WHERE user_cn = ? "
```

Comment:
This is fine (don't change it), but just wanted to mention you can do this by passing in your own TimeSource when constructing the service, then you can pretend a day has passed

Reply:
Oh cool thanks!